### PR TITLE
[ONNX] Benchmark refactored ONNX export

### PR DESCRIPTION
**Pull request by titaiwangms**
Reuse torch.onnx.export with torch_onnx patch to test ExportedProgram -> ONNX IR exporter

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -1522,6 +1527,14 @@ def forward(self, *args):
 
             model = WrapperModel(model, list(example_inputs.keys()))
 
+        if self.use_experimental_patch:
+            import torch_onnx
+
+            torch_onnx.patch_torch(error_report=True, profile=True)
+        elif "torch_onnx" in sys.modules:
+            # make sure the patch is not in effect
+            torch_onnx.unpatch_torch()
```
Comment (justinchuby) [inline benchmarks/dynamo/common.py:1536]: Is torch_onnx in namespace here? I assume so with sys.modules? Just double checking

File: benchmarks/dynamo/common.py [hunk 2]
```diff
@@ -1287,9 +1287,14 @@ def __init__(
         example_inputs,
         dynamic_shapes: bool,
         copy_before_export: bool = False,
+        use_experimental_patch: bool = False,
```
Comment (justinchuby) [inline benchmarks/dynamo/common.py:1290]: Would be good to comment on what this option does

**Review by justinchuby — APPROVED** [review]
Thanks!

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (titaiwangms) [reply benchmarks/dynamo/common.py:1536]: It looks working locally, but we can also importing it anyway, but patching/unpatching with if/else?

Comment (justinchuby) [reply benchmarks/dynamo/common.py:1536]: We can maybe do

```
try:
    import torch_onnx
    torch_onnx.unpatch_torch()
except ImportError:
    pass
```

**Pull request closed without merging**